### PR TITLE
Add status subcommands and basic e2e test based on ginkgo

### PR DESCRIPTION
**Pull request by JAORMX**
This adds a basic test that checks that a policy is able to be installed
and reports an appropriate status. This brings in all the dependencies
from ginkgo.

On the other hand, to facilitate these tests and the overall UX. This also
adds the status and is-ready subcommands. These help admins get the status
of the policies and selinuxd itself.

### CODE REVIEW

File: cmd/is-ready.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+// nolint:gochecknoinits
+func init() {
+	rootCmd.AddCommand(isreadyCmd)
+	defineIsReadyFlags(isreadyCmd)
+}
+
+func defineIsReadyFlags(rootCmd *cobra.Command) {
+	rootCmd.Flags().String("socket-path", daemon.DefaultUnixSockAddr, "The path to the socket to listen at")
```
Comment (jhrozek) [inline cmd/is-ready.go:45]: Nit: the description is wrong, it should say something like 'the path where selinuxd socket is listening at"

File: cmd/is-ready.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		fmt.Fprintf(os.Stderr, "Parsing flags: %s", err)
+		syscall.Exit(1)
+	}
+
+	httpc := getHTTPClient(opts.Path)
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTimeout)
+	defer cancel()
+
+	readyurl := "http://unix/ready/"
```
Comment (jhrozek) [inline cmd/is-ready.go:72]: What do you think about having the endpoints as exported constants of the deamon and then the commands would just prepend `http://unix/` and the endpoint they are connecting to?

File: cmd/is-ready.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+	var status map[string]bool
+	err = json.NewDecoder(response.Body).Decode(&status)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Decoding ready endpoint response: %s", err)
+		syscall.Exit(1)
+	}
+
+	if status["ready"] {
+		fmt.Fprint(os.Stdout, "yes")
```
Comment (jhrozek) [inline cmd/is-ready.go:95]: Do you think the command would be more useful in scripts if it returned 0 only if the deamon is ready?

File: cmd/status.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+}
+
+// nolint:gochecknoinits
+func init() {
+	rootCmd.AddCommand(statusCmd)
+	defineStatusFlags(statusCmd)
+}
+
+func defineStatusFlags(rootCmd *cobra.Command) {
+	rootCmd.Flags().String("socket-path", daemon.DefaultUnixSockAddr, "The path to the socket to listen at")
```
Comment (jhrozek) [inline cmd/status.go:50]: same comment about the description as above

File: tests/e2e/helpers.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+)
+
+var (
+	selinuxdInAContainer  bool
+	selinuxdContainerName string
+)
+
+const (
+	// timeout for selinuxd to report it's ready
+	selinuxdReadyTimeout float64 = 120
```
Comment (jhrozek) [inline tests/e2e/helpers.go:25]: Let's see if this is enough, I've seen upwards of 3 minutes in my k8s environment..

**Review by jhrozek — COMMENTED** [review]
some nits inline, overall looks good

File: cmd/is-ready.go (hunk 3 above)
Comment (JAORMX) [reply cmd/is-ready.go:95]: I honestly don't know... I guess we could define multiple return statuses:
* 0 -> ready
* 1 -> not ready
* 2 -> another error

What do you think?

File: tests/e2e/helpers.go (hunk 5 above)
Comment (JAORMX) [reply tests/e2e/helpers.go]: done.

File: cmd/status.go (hunk 4 above)
Comment (JAORMX) [reply cmd/status.go]: Done.

File: cmd/is-ready.go (hunk 3 above)
Comment (jhrozek) [reply cmd/is-ready.go:95]: I guess leave it for now, it's not like returncodes are API..

Comment (JAORMX) [reply cmd/is-ready.go:95]: @jhrozek OK; should we open a bug for this so we revisit it later?

**Comment by jhrozek** [comment]
/retest
there was a 503 error returned from vagrant hub

**Pull request merged**